### ISSUE TITLE
create serviceMonitor for network-costs

### ISSUE THREAD

**Issue by jessegoodier** [kubecost/cost-analyzer-helm-chart#1709]
**What problem are you trying to solve?**  

kubecost have a serviceMonitor for kubecost metrics, but not for the daemonset.

FYI: the scrapeConfig does work

**To Reproduce**  

there is no serviceMonitor in the repo for network-costs

**Expected behavior**  
create network cost service monitor similar to:
https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/templates/cost-analyzer-servicemonitor-template.yaml

**Describe the solution you'd like**  

helm values

**Describe alternatives you've considered**  

https://github.com/kubecost/poc-common-configurations/blob/main/custom-tsdb/network-costs-serviceMonitor.yaml

**How would users interact with this feature?**
helm:

```
networkCosts:
  serviceMonitor: 
    enabled: true
```

**Comment by mbolt35** [comment]
Hey @jessegoodier - I think this is because we don't want the functionality from a service monitor when it comes to network costs. A service in this case would only extract metrics from one of the N  daemonset network costs pods. We have a `PodMonitor` template though: 

https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/templates/cost-analyzer-network-costs-podmonitor-template.yaml

I think a ServiceMonitor for network-costs will cause problems since it's basically load-balancing scrape requests. There _is_ a way that we have setup to create a service for network-costs, but it requires that the network-costs setup be removed from the values.yaml and a special configuration has to be used with Prometheus to ensure that it uses "endpoint" scraping (Service creates N endpoints, all endpoints are scraped). There may be some way to configure the endpoint scrapes AND a service monitor? In general though, we just want to ensure that prometheus is always scraping all of the network costs pods AND if someone is using prom operator, it's often just easier to use the pod monitor instead of a service monitor.

**Comment by mbolt35** [comment]
Just so it's clear, there are basically two ways you can setup scraping for network-cost pods. The _default_ way in our current helm chart is with this prometheus "extra" config:

https://github.com/kubecost/cost-analyzer-helm-chart/blob/9c93abb6ba432d95e84f787c3000c38c5b5414b8/cost-analyzer/values.yaml#L449-L456

(Note line 41 - `sd_role: pod`) indicating that it will match pods to scrape based on the name. There's no service to create, just create the daemonset, update prom config, and it works. 

The next method is the explicitly remove the default behavior (ie: do _not_ use the config above), and set the `prometheusScrape` flag to true:

https://github.com/kubecost/cost-analyzer-helm-chart/blob/9c93abb6ba432d95e84f787c3000c38c5b5414b8/cost-analyzer/values.yaml#L559-L563

If you make the changes mentioned above, then yeah, I think adding a `ServiceMonitor` is correct (`prometheusScrape: true` will create a `Service`)

**Comment by jessegoodier** [comment]
I should have added more detail. 

This is _only_ for bring your own prometheus.

Our Prometheus does not look for serviceMonitors, even if the CRD is installed on the cluster because we aren't using the operator.

If a user were to set 
```
serviceMonitor:
  enabled: true
  networkCosts:
    enabled: true
```

they would get:

```
Error: UPGRADE FAILED: resource mapping not found for name: "kubecost-cost-analyzer" namespace: "" from "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"
ensure CRDs are installed first
```

Regarding the type of serviceMonitor to scrape every pod of a DS, I verified on a two-node cluster that it is scraping both:
![image](https://user-images.githubusercontent.com/31039225/191736550-c7bd9d11-7855-4a03-ad95-929c74e13258.png)


Looking at an example from the prometheus serviceMonitor for the node-exporter DS, the current spec looks right by my eye.

Pushed a commit with a comment explaining the serviceMonitor in values.yaml.

Also pushing a PR on howto here: https://github.com/kubecost/poc-common-configurations/pull/31/files#diff-db46239f8b53d07c0516b432511b56fdb728d8d264f9a09f345e5561e52abd7eL8
